### PR TITLE
merge Feature/interrupts to develop

### PR DESCRIPTION
**Pull request by ShonGutman**

### CODE REVIEW

File: cpu/interrupts/irq.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#include "drivers/ports.h"
+
+#define TOTAL_IRQ 16
+
+#define MASTER_PIC_COMMAND_PORT 0x20
+#define SLAVE_PIC_COMMAND_PORT 0xa0
+
+#define EOI 0x20
+
+void (*irq_routines[TOTAL_IRQ])(interrupt_registers_struct) =
```
Comment (Tedy2606) [inline cpu/interrupts/irq.c:11]: did not understand what this is

File: cpu/interrupts/irq.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+void (*irq_routines[TOTAL_IRQ])(interrupt_registers_struct) = 
+{
+    0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0
+};
+
+void irq_handler(interrupt_registers_struct regs)
+{
+    /** After every interrupt we need to send and EOI (end of interrupt) to the PICs
+     * or they will not send another interrupt again
```
Comment (Tedy2606) [inline cpu/interrupts/irq.c:20]: replace "or" with "so"

File: cpu/interrupts/isr.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
+#include "isr.h"
+#include "idt.h"
+#include "pic.h"
+#include "drivers/screen.h"
+
+void isr_handler(interrupt_registers_struct regs)
+{
+    print("interrupt occured! system hlts");
+    for(;;);
```
Comment (Tedy2606) [inline cpu/interrupts/isr.c:9]: Please replace this with a While(true), I know it has the same functionality but this is not how the for loops are commonly used

File: cpu/interrupts/idt.c [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    idt_entries[num].segmentSelector = GDT_CODE_SEGMENT_SELECTOR;
+    idt_entries[num].flags = FLAGS;
+}
+
+void idt_init()
+{
+    idt_info.size = sizeof(idt_entry_struct) * IDT_ENTRIES - 1;
+    idt_info.offset = (uint32)&idt_entries;
+
+    __asm__ ("lidt (%0)" : : "r"(&idt_info));
```
Comment (Tedy2606) [inline cpu/interrupts/idt.c:26]: add a comment explaining what this line does. Other than that everything looks good

File: cpu/interrupts/irq.c (hunk 1 above)
Comment (ShonGutman) [reply cpu/interrupts/irq.c:12]: done

File: cpu/interrupts/idt.c (hunk 4 above)
Comment (ShonGutman) [reply cpu/interrupts/idt.c:27]: done

**Review by Tedy2606 — COMMENTED** [review]
LGTM

**Review by Tedy2606 — APPROVED** [review]
LGTM

**Pull request merged**